operations/configuration_management/Write_Ansible_Code_for_CFS.md

```diff
@@ -11,22 +11,55 @@ Help with Ansible can be found in the external Ansible documentation:
 
-For example, `hosts: Compute` would be used to target the compute nodes. Users can create additional sections that target other node types, or adjust the hosts that the included roles will run against as necessary. It is also possible to target multiple groups within a section of a playbook, or to specify complex targets, such as nodes that are in one group and not in another group. The syntax for this is available in the external [Ansible documentation](https://docs.ansible.com/ansible/latest/user_guide/intro_patterns.html#common-patterns). Hosts can be in more than one group at a time if there are user-defined groups. In this case, Ansible will run all sections that match the node type against the node.
+For example, `hosts: Compute` would be used to target the compute nodes. Users can create additional sections that target other node types, or adjust the hosts that the included roles will run against as necessary. It is also possible to target multiple groups within a section of a playbook, or to specify complex targets, such as nodes that are in one group and not in another group. The syntax for this is available in the external [Ansible documentation](https://docs.ansible.com/ansible/latest/user_guide/intro_patterns.html#common-patterns). Hosts can also be in more than one group at a time. In this case, Ansible will run all sections that match the node type against the node.
 
 See the [Ansible Inventory](Ansible_Inventory.md) section for more information about groups that are made automatically available through CFS dynamic inventory.
 
 ## Performance and Scaling Tips
 
 CFS will handle scaling up Ansible to run on many hosts, but there are still places where performance can be improved by correctly writing Ansible plays.
 
-* Use image customization when possible to limit how many times a task is run and improve boot times. Configuration that is the same for all nodes of the same type will benefit from image customization. See the next section for how to target specific tasks for running only during image customization.
-* Import roles rather than playbooks. Each time a new playbook starts, Ansible automatically gathers facts for all the systems it is running against. This is not necessary more than once and can slow down Ansible execution.
+### Using Image Customization
+
+* Use image customization when possible to limit how many times a task is run and improve boot times. Configuration that is the same for all nodes of the same type will benefit from image customization.
+* Use different playbooks for image-customization and node-personalization.  Moving image customizations tasks to their own playbook can remove the need to evaluate that conditionals in a shared playbook, as well as ensuring that tasks are not accidentally running in both modes needlessly.
+* See [Target Ansible Tasks for Image Customization](Target_Ansible_Tasks_for_Image_Customization.md) for more information on writing for image customization.
+
```

Comment:
Maybe add something about avoid interacting with system services that are not yet running. I've seen a number of people that are confused when they attempt to enable a service using image configuration but systemd isn't available. Its neither a performance nor a scaling tip, so if there is a more applicable section....

Reply:
There's a section elsewhere on "## Image Customization Limitations" which already contains "However, the image is not a running node, so actions that require a running system, such as starting/reloading a service, will not work properly and will cause the Ansible play to fail."